template.tpl

```diff
@@ -436,7 +505,165 @@ ___WEB_PERMISSIONS___
 
 ___TESTS___
 
-scenarios: []
+scenarios:
+- name: Test injectScript
+  code: |-
+    var success, failure;
+    mock('injectScript', (url, onSuccess, onFailure) => {
+      success = onSuccess;
+      failure = onFailure;
+      assertThat(url).isEqualTo('https://www.redditstatic.com/ads/pixel.js');
+      onSuccess();
+    });
+
+
+    // Call runCode to run the template's code.
+    runCode(mockData);
+
+    // Verify that the tag finished successfully.
+    assertApi('injectScript').wasCalledWith(url, success, failure, 'rdtPixel');
+    assertApi('gtmOnSuccess').wasCalled();
+- name: Test rdt exists
+  code: |
+    mock('copyFromWindow', key => {
+      if (key === 'rdt') return () => {};
+    });
+
+    // Call runCode to run the template's code.
+    runCode(mockData);
+
+    // Verify that the tag finished successfully.
+    assertApi('gtmOnSuccess').wasCalled();
+- name: Test rdt doesn't exist
+  code: |-
+    let rdt;
+
+    mock('copyFromWindow', key => {
+      if (key === 'rdt') return rdt;
+    });
+
+    mock('createQueue', function(name) {
+      assertThat(name).isEqualTo('rdt.callQueue');
+      return function(item) {
+        assertThat(item).isNotNull();
+      };
+    });
+
+    mock('setInWindow', (key, val) => {
+      if (key === 'rdt') rdt = val;
+    });
+
+    // Call runCode to run the template's code.
+    runCode(mockData);
+
+    // Verify that the tag finished successfully.
+    assertApi('setInWindow').wasCalled();
+    assertApi('gtmOnSuccess').wasCalled();
+- name: Test enabledFirstPartyCookies false
+  code: |-
+    mockData.enableFirstPartyCookies = false;
+
+    let isSet = false;
+
+    mock('copyFromWindow', key => {
+      if (key === 'rdt') return function() {
+         if (arguments[0] === 'disableFirstPartyCookies') {
+           isSet = true;
+         }
+      };
+    });
+
+    // Call runCode to run the template's code.
+    runCode(mockData);
+
+    // Verify that the tag finished successfully.
+    assertThat(isSet, 'disableFirstPartyCookies not set even though first party cookies are disabled').isTrue();
+    assertApi('gtmOnSuccess').wasCalled();
+- name: Test enabledFirstPartyCookies true
+  code: |-
+    mockData.enableFirstPartyCookies = true;
+
+    mock('copyFromWindow', key => {
+      if (key === 'rdt') return function() {
+         if (arguments[0] === 'disableFirstPartyCookies') fail('disableFirstPartyCookies called even though first party cookies are enabled');
+      };
+    });
+
+    // Call runCode to run the template's code.
+    runCode(mockData);
+
+    // Verify that the tag finished successfully.
+    assertApi('gtmOnSuccess').wasCalled();
+- name: Test Advanced Matching
+  code: |
+    mockData = {
+      id: "t2_123",
+      event_type: "PageVisit",
+      enableFirstPartyCookies: true,
+      advancedMatching: true,
+      advancedMatchingParams: [
+        {name: 'optOut', value: 1},
```

Comment:
I believe this is actually a boolean field that's converted to a 1 or 0?

https://github.snooguts.net/reddit/reddit-conversion-pixel-js/blob/develop/src/lib/pixelLib.js#L93

Reply:
The value of optOut can be any [truthy](https://developer.mozilla.org/en-US/docs/Glossary/Truthy) or [falsy](https://developer.mozilla.org/en-US/docs/Glossary/Falsy) value.  If optOut is not specified, it defaults to false (and permits retargeting).